EchoPro/survey.py

```diff
@@ -327,269 +328,502 @@ def read_validated_data( self ,
+        # Indexed / organized by sex , station, and stratum
+        length_proportion_table = (
+            station_length_aggregate
+            .pivot_table( columns = [ 'group' , 'station' , 'stratum_num' ] , 
+                         index = [ 'length_bin' ] ,
+                         values = [ 'p_within_station' ] )[ 'p_within_station' ]
+        )
+        
+        ### Calculate combined station fraction means
+        # Station 1 
+        stn_1_fraction = ( sex_stn_prop_merged.loc[ 1 , ( 'p_stn' ) ]                           
+                          / ( sex_stn_prop_merged.loc[ 1 , ( 'p_stn' ) ] 
+                            + sex_stn_prop_merged.loc[ 2 , ( 'p_sex_stn' ) ] ) )
+        
+        # Station 2
+        stn_2_fraction = ( sex_stn_prop_merged.loc[ 2 , ( 'p_sex_stn' ) ]                          
+                          / ( stn_1_fraction
+                            + sex_stn_prop_merged.loc[ 2 , ( 'p_sex_stn' ) ] ) )
+        
+        ### Calculate the average weight across all animals, males, and females
+        # Pull fitted weight values
+        fitted_weight = self.statistics['length_weight']['length_weight_df']
+        
+        # Total
+        total_weighted_values = ( length_proportion_table.loc[ : , ( 'all' , 1 ) ] * station_proportions.loc[ 1 , ] +
+                                  length_proportion_table.loc[ : , ( 'all' , 2 ) ] * station_proportions.loc[ 2 , ] )
+        total_weight = fitted_weight[ 'modeled_weight' ].dot( total_weighted_values.reset_index( drop = True ) )
+        
+        # Male
+        male_weighted_values = ( length_proportion_table.loc[ : , ( 'male' , 1 ) ] * stn_1_fraction.loc[ 'male' ] +
+                                 length_proportion_table.loc[ : , ( 'male' , 2 ) ] * stn_2_fraction.loc[ 'male' ] )
+        male_weight = fitted_weight[ 'modeled_weight' ].dot( male_weighted_values.reset_index( drop = True ) )
+        
+        # Female
+        female_weighted_values = ( length_proportion_table.loc[ : , ( 'female' , 1 ) ] * stn_1_fraction.loc[ 'female' ] +
+                                   length_proportion_table.loc[ : , ( 'female' , 2 ) ] * stn_2_fraction.loc[ 'female' ] )
+        female_weight = fitted_weight[ 'modeled_weight' ].dot( female_weighted_values.reset_index( drop = True ) )
+        
+        ### Store the data frame in an accessible location
+        self.biology[ 'weight' ][ 'weight_strata_df' ] = pd.DataFrame( {
+            'stratum': total_weight.index ,
+            'female_proportion': sex_proportions.loc[ 'female' , : ][ 'p_overall_station' ].reset_index(drop=True) ,
+            'male_proportion': sex_proportions.loc[ 'male' , : ][ 'p_overall_station' ].reset_index(drop=True) ,
+            'female_average_weight': female_weight ,            
+            'male_average_weight': male_weight ,
+            'total_average_weight': total_weight
+        } )
```

Comment:
Following the above "noun_modifier" comment, maybe these become `proportion_male/female` and `average_weight_male/female/total`?